cli/tests/activate.bats

```diff
@@ -2228,7 +2438,7 @@ EOF
 # ---------------------------------------------------------------------------- #
 
 # bats test_tags=activate,activate:nested_flox_activate_tracelevel
-@test "{bash,fish,tcsh,zsh}: confirm _flox_activate_tracelevel set in nested activation" {
```

Comment:
suggestion(nb): not sure if this one needed to be split up since the tests are almost identical now.

can we refactor this into a generic function that is then called by four tests with specific arguments for the shell it runs with instead?


```bats
function confirm_tracelevel_generic() {
  shell=${1?}
  config=${2?}

  cat >"$HOME/$config" <<EOF
touch "$PROJECT_DIR/_flox_activate_tracelevel.in_test"
test -n "\$_flox_activate_tracelevel" || touch "$PROJECT_DIR/_flox_activate_tracelevel.not_defined"
EOF
  # Finish by appending shell-specific flox activation syntax.
  echo "eval \"\$($FLOX_BIN activate --dir $PROJECT_DIR)\"" >>"$HOME/$config"

  # Create a test environment.
  _temp_env="$(mktemp -d)"
  "$FLOX_BIN" init -d "$_temp_env"

  # Activate the test environment, which will launch an interactive shell that
  # sources the relevant dotfile.

  FLOX_SHELL="$shell" USER="$REAL_USER" NO_COLOR=1 run -0 expect "$TESTS_DIR/activate/activate.exp" "$_temp_env"
  refute_output --partial "_flox_activate_tracelevel not defined"
  run rm "$PROJECT_DIR/_flox_activate_tracelevel.in_test"
  assert_success
  run rm "$PROJECT_DIR/_flox_activate_tracelevel.not_defined"
  assert_failure

  rm -rf "$_temp_env"
}

```

Reply:
Tweaked a few things but took that approach